analyses/compare-gistic/util/GISTIC-comparison-functions.R

```diff
@@ -14,64 +14,51 @@
 
 #### Implemented in both `01-GISTIC-cohort-vs-histology.Rmd` and `02-GISTIC-tidy-data-prep.Rmd`
 format_gistic_genes <- function(genes_file,
-                                include_peak_info = FALSE) {
+                                include_peak_info = TRUE) {
   # Given the GISTIC result `amp_genes_conf_90.txt` or `del_genes_conf_90.txt`
   # files for the entire cohort or a specific histology, get the vector/data.frame
   # of genes for each amplification/deletion.
   #
   # Args:
   #  genes_file: file path to the `amp_genes.conf_90.txt` or
   #              `del_genes.conf_90.txt` file
-  #   include_peak_info: binary flag indicating weather or not to include the
-  #                      detection peak info
   #
   # Return:
   #  genes_output: a vector/data.frame with all the genes included in the file
-
-  genes_ragged_df <- data.table::fread(genes_file,
-    data.table = FALSE
-  )
-  genes_list <- as.list(genes_ragged_df)
-  genes_list <- genes_list %>%
-    # This removes any element from the list that is all NA -- most likely a
-    # result of reading in a ragged array
-    purrr::discard(~ all(is.na(.))) %>%
-    # This removes the element of the list that is essentially the "header" for
-    # the file
-    purrr::discard(~ any(str_detect(., "cytoband|q value"))) %>%
-    # Remove blanks -- result of ragged data.frame
-    purrr::modify(~ .[. != ""])
-
-  if (include_peak_info == TRUE) {
-    genes_list <- genes_list %>%
-      # Remove everything before and including the wide peak boundaries for each
-      # remaining element of the list
-      purrr::modify(~ .[-c(1:(str_which(., "chr")) - 1)])
-
-    # This will give us the data.frame of all the genes that were included
-    genes_output <-
-      data.table::rbindlist(lapply(genes_list, function(x)
-        data.frame(t(x))),
-      fill = TRUE
-      )
-
-    # Tidy the data.frame into a key and value format where the detection peak
-    # region is the key and the genes in that region are the values
+  
+  genes_ragged_list <- data.table::fread(genes_file,
+                                         data.table = FALSE)
+  
+  # Transpose data
+  genes_transposed <- genes_ragged_list %>%
+    t()
+  
+  # Make the header information in the first row the column names
+  colnames(genes_transposed) <- genes_transposed[1,]
+  
+  genes_df <- genes_transposed %>%
+    # Make into a data.frame object
+    as.data.frame() %>%
```

Comment:
Can you add a `stringsAsFactor = FALSE` in here? It is turning these things into Factors and when you do the `gather` step at line 47 it says it's dropping values that don't fit into those attributes. May be best to tell R to avoid coercing these to factors.